src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RTSocketConnection.cs

```diff
@@ -258,6 +259,10 @@ protected override void ShutdownCore(TimeSpan timeout)
                     throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(exceptionToThrow);
                 }
             }
+            catch (OperationCanceledException)
+            {
+                throw;
```

Comment:
This was a bug fix for an issue I found while implementing this. If the sendCts.Token has been cancelled, the writeTask will throw an OperationCanceledException when the result is fetched. This was getting caught by the general Exception catch block presuming it was an exception from the UWP StreamSocket class. If this does get thrown, abortWrite will not get set to false so in the finally block AbortWrite() will get called.